### ISSUE TITLE
SSH key is not added to the agent with missing error message

### ISSUE THREAD

**Issue by horror-vacui** [keepassxreboot/keepassxc#7152]
## Overview
SSH key is not added to the agent. No error message, neither in the GUI or in the terminal, just a red drop-down stripe, which I assume supposed to be the background for the error message. 

## Steps to Reproduce
1. Open KeepassXC
2. Right-click on an entry with an SSH key and select "Add key to SSH Agent"

## Expected Behavior
Key is added to the agent.
If there is an error, the message should be readable.

## Actual Behavior
Key isn't added to the Agent, and a red stripe appears, and after a few seconds disappear. See screenshot.
![KeepassXC_ssh_error](https://user-images.githubusercontent.com/31111509/142774020-15f90511-5c49-4d80-8542-dbf33668def6.png)
![KeepassXC_SSHAgent_connection_working](https://user-images.githubusercontent.com/31111509/142774022-290c8b8b-9426-4492-b135-15e4a1dd128f.png)

PS: I am not sure how to enable the debug mode from the GUI. The man page does mention it either.

## Context
Connection to the SSH Agent is working. `ssh-add -L` does not list the key.

KeePassXC - Version 2.6.6
Revision: 9c108b9

Qt 5.15.2
Debugging mode is disabled.

Operating system: Manjaro Linux
CPU architecture: x86_64
Kernel: linux 5.15.2-2-MANJARO

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- libgcrypt 1.9.4-unknown

Desktop Env: KDE
Windowing System: X11

**Comment by hifi** [comment]
Which service created the socket, `ssh-agent`, keyring daemon or `gpg-agent`? Does that socket match your terminal environment where you use `ssh-add`?

Blank error message is weird, though.

**Comment by horror-vacui** [comment]
I am a bit overwhelmed here. I am using Linux for 15 years, but I am no IT guy by profession. Sorry. Are the information below of any help?
KeepassXC behaves the same way whether I start it from the menu or from the terminal.

```
zoltan@adrian ~ $ echo $SSH_AGENT_SOCK

zoltan@adrian ~ $ echo $SSH_AUTH_SOCK
/run/user/1000/ssh-agent.socket
zoltan@adrian ~ $ ps -aux | grep ssh
root         882  0.0  0.0   8976  5764 ?        Ss   14:51   0:00 sshd: /usr/bin/sshd -D [listener] 0 of 10-100 startups
zoltan      1183  0.0  0.0   5952  3904 ?        Ss   14:51   0:00 /usr/bin/ssh-agent -D -a /run/user/1000/ssh-agent.socket
zoltan@adrian ~ $ ps -aux | grep gpg
zoltan     17802  0.0  0.0  81516  3832 ?        SLs  21:32   0:00 /usr/bin/gpg-agent --supervised
```

I do not know how to get some error and warning messages into the console or into a log. Therefore I could not even search the internet for similar cases.

There is KDE Wallet running in the system with ksshaskpass installed. I have not tinkered with them, or with any agent. I surmise from KeepassXC FAQ, that the ksshaskpass is needed for confirmation of using the keys (supplying the passphrase for them?).

**Comment by hifi** [comment]
Okay, that's enough formation to know the socket is created by ssh-agent, thanks.

What type the key you have is and can you possibly try a snapshot from https://snapshot.keepassxc.org/latest/ (AppImage) to see if the issues exists in the development version?

**Comment by horror-vacui** [comment]
I was able to look into it again in detail, and I can confirm that  I am to blame about the ssh-key addition issue: the keys were were as attachments, but they were not select under the "SSH Agent" tab of the database entry. Basically I forgot to tell Keepass that I should select which attachment is to be treated as the private key. 

On the other hand I identified two things in KeepassXC, which have mislead me, and I believe they could be improved:
1. The lack of any error message, especially with the red drop-down stripe. There was nothing in the terminal, and I have not found any logs.
2. The menu after right-clicking on a database entry (or item? sorry, my vocabulary has forsaken me) has clearly listed the "Add key to SSH Agent" option, even though no keys were associated with the entry, just attachments were added to it. This gave me the false conviction that I set the up keys correctly.

PS: I've tested with the latest appimage, but after noticing that it behaves the same way as version 2.6.6, I went back to the 2.6.6. for debugging.
